### ISSUE TITLE
Tenant seeders fail during in-memory SQLite tests

### ISSUE THREAD

**Issue by kylekz** [hyn/multi-tenant#906]
Following this example: https://github.com/tenancy/multi-tenant/pull/656

During tenant creation via a tenant seeder, I create a role, some permissions, and attach those together. This works during normal, out-of-test use.

I have a migration that converts my ACL system from spatie/laravel-permissions to laratrust, which involves adding a new column named `display_name`. When running tests, I get the following error:

```
Illuminate\Database\QueryException : SQLSTATE[HY000]: General error: 1 table permissions has no column named display_name (SQL: insert into "permissions" ("name", "display_name", "updated_at", "created_at") values (view-admin, View Admin, 2020-03-10 03:34:45, 2020-03-10 03:34:45))
```

If I don't insert a `display_name` and dump the result model, I get the correct model back but with no `display_name` attribute, not even a null value. I can dump the tenant migrations table and very clearly see the corresponding migration going through. I get the same result when I attempt to do any of this ACL seeding during the actual test.

What has me confused is that it looks like the schema has the following columns: `id, name, created_at, updated_at`
Whereas spatie/permission has: `id, name, guard_name, created_at, updated_at`
And Laratrust/my intended has: `id, name, display_name, type, created_at, updated_at`

So I'm not sure where this broken schema is coming from or how it's happening.

At first I attempted running a `tenancy:db:seed`, however that fails with a `table permissions does not exist` error, which I assume is to do with `migrate` disconnecting, thus nuking the in-memory database.

**Comment by ArlonAntonius** [comment]
Hi there,

- Is the Tenant DB also in memory?
- Can you doublecheck that it's using the right model?
- Can you check if the table exists with the right columns?
- Is the same table also on the system connection? If so, can you check that the model is using the right connection?

**Comment by kylekz** [comment]
> Is the Tenant DB also in memory?

I believe so, the linked MR and corresponding MR within that (https://github.com/tenancy/multi-tenant/pull/627#issuecomment-427492644) show config options to use, which include the system DB within memory.

>Can you doublecheck that it's using the right model?

Yes, including dumping the result of `get_class($permission)` and so on. It's using the `UsesTenantConnection` trait. `$permission->getConnectionName()` returns `tenant`.

>Can you check if the table exists with the right columns?

`DB::connection('tenant')->table('permissions')->columns` results in `NULL`

`Schema::connection('tenant')->getColumnListing('permissions')` results in `[id, name, created_at, updated_at]`, which doesn't make sense as at no point has the schema been that.

>Is the same table also on the system connection? If so, can you check that the model is using the right connection?

No, it's strictly a tenant table.

After figuring out that `getColumnListing()` dumps the table structure properly, I've actually figured out the cause. It seems like SQLite silently exits after dropping a column. Consider this migration:

```
Schema::table('permissions', function (Blueprint $table) {
    $table->dropColumn('guard_name');
    $table->string('display_name')->nullable();
    $table->tinyInteger('type')->default(PermissionType::GLOBAL()->getValue());
 });
```

`guard_name` gets dropped, but the rest of the migration doesn't go through, resulting in a broken schema. If I split that one closure into two like so, it works:

```
Schema::table('permissions', function (Blueprint $table) {
    $table->dropColumn('guard_name');
 });
Schema::table('permissions', function (Blueprint $table) {
    $table->string('display_name')->nullable();
    $table->tinyInteger('type')->default(PermissionType::GLOBAL()->getValue());
 });
```

It's kinda gross, but I guess it's the solution. I've always written my migrations where I have a single file for an entire operation, including data transfer/mutation, but from now on I suppose I'll have to split drops from creates in separate migrations for the sake of keeping things clean. if you have any recommendations on potentially fixing the root SQLite behavior instead of fixing migrations, let me know.

Might be worth having a write up on having tests for multitenancy. I've spent so long trying to figure all this out, and I've finally got basic unit tests from 7s down to 700ms. If there's a docs repo then I could do a PR once I have some time.
